vault/auth.go

```diff
@@ -407,7 +407,7 @@ func (c *Core) remountCredential(ctx context.Context, src, dst namespace.MountPa
 	}
 
 	c.authLock.Lock()
-	if match := c.router.MountConflict(ctx, dstRelativePath); match != "" {
+	if match := c.router.MountConflict(ctx, dstRelativePath); match != dst.Namespace.Path {
 		c.authLock.Unlock()
 		return fmt.Errorf("path in use at %q", match)
 	}
```

Comment:
Maybe I'm thinking about this wrong, but shouldn't storage migration happen while the route is tainted, so we don't get any new requests?

Line 415 executes before the router move in 431 that has been hooked, and so we risk new requests coming in while storage is being moved, which may result in weird happenstances when e.g., entries that should exist don't?

My 2c.

Reply:
I believe the `routeEntry` is still tainted at that point (old path tainted in 397, new path untainted in 524), so I think it should not be possible that new requests come in in the meantime. But I agree, it's weird that the `MountEntry` is untainted at that point. I'll try to find out why that might be.